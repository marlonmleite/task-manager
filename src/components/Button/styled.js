import styled from 'styled-components'

export const ButtonStyled = styled.button`
  padding: 0.4375rem 0.75rem;
  line-height: 1.25rem;
  font-size: 0.875rem;
  background-color: ${({ theme, color }) => theme.colors[color]};
  color: ${({ theme, color }) => theme.button.text[color]};
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid ${({ theme, color }) => theme.colors[color]};
  border-radius: 2px;
  position: relative;

  :focus,
  :hover {
    text-decoration: none;
    background-image: linear-gradient(transparent, rgba(0,0,0,.05) 40%, rgba(0,0,0,.1));
  }

  :focus {
    outline: none;
  }

  .disabled,
  :disabled {
    opacity: .65;
    box-shadow: none;
  }

  :not([disabled]):not(.disabled) {
    cursor: pointer;
  }

  ${({ loading }) => loading && `
    > span {
      visibility: hidden;
    }
  `};
`

export const LoadingContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
