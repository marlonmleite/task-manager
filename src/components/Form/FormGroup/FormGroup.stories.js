import React from 'react'
import { storiesOf } from '@storybook/react'
import FormGroup from './'
import { Input } from '../Input'

const stories = storiesOf('FormGroup', module)

stories.add('default', () => (
  <div>
    <div>
      <FormGroup name="firstName" label="First name">
        <Input id="firstName" defaultValue="" />
      </FormGroup>
    </div>
    <div>
      <FormGroup name="lastName" label="Last name">
        <Input id="lastName" defaultValue="Ferreira" />
      </FormGroup>
    </div>
  </div>
))

stories.add('invalid input', () => (
  <div>
    <div>
      <FormGroup name="email" label="Email" error="Enter a email" invalid>
        <Input id="email" defaultValue="" />
      </FormGroup>
    </div>
    <div>
      <FormGroup name="email" label="Email" error="Invalid email" invalid>
        <Input id="email" defaultValue="marlon@" />
      </FormGroup>
    </div>
  </div>
))
