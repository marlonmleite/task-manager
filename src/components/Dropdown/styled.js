import styled, { css } from 'styled-components'

export const Dropdown = styled.div`
  position: relative;

  > button {
    width: 100%;
  }
`

export const DropdownContent = styled.div`
  position: absolute;
  min-width: 200px;
  color: #89898a;
  background-color: #fff;
  border-radius: 2px;
  margin-top: 2px;
  padding: 3px 5px;
  top: 100%;
  left: 0;
  z-index: 99;
  display: flex;
  flex-direction: column;
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.14), 0 3px 14px 2px rgba(0,0,0,0.12), 0 5px 5px -3px rgba(0,0,0,0.2);

  ${({ right }) => right && css`
    left: auto;
    right: 0;
  `};
`

export const DropdownButton = styled.div`
  display: flex;
  user-select: none;
  align-items: center;
  justify-content: space-between;

  > span {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`

export const DropdownOption = styled.div`
  width: 100%;
  position: relative;
  padding: 0 1.25rem 0 1.5rem;
  box-sizing: border-box;
  color: #222;
  height: 36px;
  line-height: 36px;
  font-size: 14px;

  :hover {
    background-color: #eee;
    cursor: pointer;
  }

  ${({ selected }) => selected && css`
    :before {
      content: '';
      border-left: 2px solid #666;
      border-bottom: 2px solid #666;
      width: 10px;
      height: 6px;
      margin-top: -7px;
      position: absolute;
      left: 5px;
      top: 50%;
      transform: rotate(-45deg);
    }
  `};
`

export const CaretStyled = styled.div`
  margin-left: 10px;
  width: 0px;
  height: 0px;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid ${({ color, theme }) => theme.button.text[color]};
`
