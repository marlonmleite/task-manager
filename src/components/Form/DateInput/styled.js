import styled from 'styled-components'

const Date = styled.div`
  .react-datepicker {
    border-radius: 2px;
    font-family: inherit;
  }
`

export default Date
