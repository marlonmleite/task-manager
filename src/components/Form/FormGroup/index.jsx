import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { FormGroupStyled, ErrorStyled } from './styled'

class FormGroup extends Component {

  static defaultProps = {
    error: '',
  };

  static propTypes = {
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
    error: PropTypes.string,
  }

  render() {
    const { name, label, children, error } = this.props

    return (
      <FormGroupStyled invalid={!!error}>
        {children}
        <label htmlFor={name}>
          {label}
        </label>
        {error && (
          <ErrorStyled>
            {error}
          </ErrorStyled>
        )}
      </FormGroupStyled>
    )
  }

}

export default FormGroup
