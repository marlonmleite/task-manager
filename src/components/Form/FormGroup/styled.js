import styled, { css } from 'styled-components'

const form = {
  labelOffset: '25px',
  paddingTop: '5px',
}

const topLabel = css`
  top: calc(${form.labelOffset} - (${form.paddingTop} + 10px));
  font-size: 12px;
`

export const FormGroupStyled = styled.div`
  width: 100%;
  font-size: 15px;
  position: relative;
  display: inline-flex;
  flex-direction: column; 
  color: ${({ theme }) => theme.form.label};
  padding: ${form.labelOffset} 0 10px 0;

  label {
    position: absolute;
    display: block;
    top: calc(${form.labelOffset} + ${form.paddingTop});
    left: 0px;
    transition: 0.2s ease all;
    pointer-events: none;
    cursor: text;
  }

  input {
    :focus ~ label,
    :not([value=""]) ~ label {
      ${topLabel};
    }
  }

  select,
  div {
    :focus ~ label,
    :not(.empty) ~ label {
      ${topLabel};
    }
  }

  input,
  select {
    :focus ~ label {
      color: ${({ theme }) => theme.focus};
    }
  }

  ${({ theme, invalid }) => invalid && css`
    input,
    select,
    input:focus,
    select:focus {
      border-color: ${theme.colors.danger};
    }

    input:focus ~ label,
    select:focus ~ label {
      color: inherit;
    }
  `};
`

export const ErrorStyled = styled.span`
  color: ${({ theme }) => theme.colors.danger};
  font-size: 13px;
`
