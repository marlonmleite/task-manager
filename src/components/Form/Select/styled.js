import styled from 'styled-components'
import { InputComponent } from '../Input'

const Select = InputComponent.withComponent('select')

const SelectStyled = styled(Select)`
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
`

export default SelectStyled
