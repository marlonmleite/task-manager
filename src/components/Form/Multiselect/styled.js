import Select from 'react-select'
import styled from 'styled-components'

const SelectStyled = styled(Select)`
  .multiselect {
    &__control {
      border-color: #e5e5e5;
      border-style: solid;
      border-width: 0 0 1px 0;
      border-radius: 0;
      background-color: transparent;
      min-height: 38px;

      &--is-focused {
        box-shadow: none;
        border-color: ${({ theme }) => theme.focus};
      }
    }

    &__value-container {
      padding: 5px 5px 5px 0;
    }

    &__menu {
      border-radius: 2px;
    }
  }
`

export default SelectStyled
