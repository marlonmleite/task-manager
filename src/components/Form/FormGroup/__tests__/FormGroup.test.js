import React from 'react'
import { shallow } from 'enzyme'
import FormGroup from '../'
import { ErrorStyled } from '../styled'

describe('FormGroup', () => {
  it('should render', () => {
    const component = shallow(<FormGroup label="Abc" name="abc">a</FormGroup>)

    expect(component).toMatchSnapshot()
  })

  it('should render with error', () => {
    const component = shallow(<FormGroup label="Abc" name="abc" error="asdf">a</FormGroup>)
    const error = component.find(ErrorStyled)

    expect(error.exists()).toBeTruthy()
  })
})
