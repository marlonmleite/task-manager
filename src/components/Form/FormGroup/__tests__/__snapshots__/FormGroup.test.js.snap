// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should render 1`] = `
<styled.div
  invalid={false}
>
  a
  <label
    htmlFor="abc"
  >
    Abc
  </label>
</styled.div>
`;
